database/database.go

```diff
@@ -22,6 +27,67 @@ var (
 
 // RelayerDatabase is a key-value store for relayer state, with each blockchainID maintaining its own state
 type RelayerDatabase interface {
-	Get(blockchainID ids.ID, key []byte) ([]byte, error)
-	Put(blockchainID ids.ID, key []byte, value []byte) error
+	Get(relayerKey common.Hash, key []byte) ([]byte, error)
```

Comment:
Agreed the name `key` on it's own is no longer descriptive enough. I'm hesistant on the name `keyType`, since it is still a key and not a "type" of key. I'd prefer something like `dataKey` to indicate that it's used to access data to be returned to the caller.

As for changing it to an enum, my only hesistation there is that it makes the interface less general. It would slightly increase type safety, but we do already have consts for the single key we use.  I don't feel strongly about this though, so happy to make this change if you think it's the right call.

> we'll probably want to prepend the key type prefix or something

Yup, that's pretty much what I was thinking. It shouldn't matter so much, as long as it's consistent in `Get` and `Put`.

Reply:
> I'd prefer something like dataKey to indicate that it's used to access data to be returned to the caller.

Devils advocate here, but isn't that true of both/all DB keys? 😅 

Changing it to an enum does make the interface less general, but I think that's actually a good thing here, since it will prevent it from being used incorrectly. Add new values to the enum is trivial for any code changes, but with a generic `[]byte` type, it could be easy to confuse what value should be passed.

Reply:
On the naming front, my only strong feeling is that the name should end in "key" to denote that that parameter is exactly a key, and not some other descriptor. An alternative to `relayerKey/dataKey` would be `relayerID/key`. `relayerID` seems appropriate to me since that key is far less likely to change than the `key` in the future.

Reply:
Gotcha, I was think of the data format closer to:
```
{
    "latestBlock": {
        "<relayer_key_1>": 42,
        "<relayer_key_2>": 43
    }
}
````

Rather than:
```
{
    "<relayer_key_1>": {
        "latestBlock: 42
    },
    "<relayer_key_2>": {
        "latestBlock: 43    
    }
}
```

But I see now that neither are correct. In the local JSON DB set up, we each `relayer_key` gets it's own file, and in the external flat key value store DB, the keys will be along the lines of "latestBlock-<relayer_key>".